packages/storage/src/service.ts

```diff
@@ -16,55 +16,146 @@
  */
 
 import { FirebaseApp } from '@firebase/app-types';
-import * as args from './implementation/args';
 import { Location } from './implementation/location';
 import { FailRequest } from './implementation/failrequest';
 import { Request, makeRequest } from './implementation/request';
 import { RequestInfo } from './implementation/requestinfo';
 import { XhrIoPool } from './implementation/xhriopool';
-import { Reference } from './reference';
+import { Reference, getChild } from './reference';
 import { Provider } from '@firebase/component';
 import { FirebaseAuthInternalName } from '@firebase/auth-interop-types';
 import { FirebaseOptions } from '@firebase/app-types-exp';
 import * as constants from '../src/implementation/constants';
-import * as errorsExports from './implementation/error';
+import {
+  invalidArgument,
+  appDeleted,
+  noDefaultBucket
+} from './implementation/error';
+
+export function isUrl(path?: string): boolean {
+  return /^[A-Za-z]+:\/\//.test(path as string);
+}
+
+/**
+ * Returns a firebaseStorage.Reference for the given url.
+ */
+function refFromURL(service: StorageService, url: string): Reference {
+  return new Reference(service, url);
+}
 
 /**
- * A service that provides firebaseStorage.Reference instances.
+ * Returns a firebaseStorage.Reference for the given path in the default
+ * bucket.
+ */
+function refFromPath(
+  ref: StorageService | Reference,
+  path?: string
+): Reference | null {
+  if (ref instanceof StorageService) {
+    const service = ref;
+    if (service._bucket == null) {
+      throw noDefaultBucket();
+    }
+    const reference = new Reference(service, service._bucket!);
+    if (path != null) {
+      return refFromPath(reference, path);
+    } else {
+      return reference;
+    }
+  } else {
+    // ref is a Reference
+    if (typeof path === 'string') {
+      if (path.includes('..')) {
+        throw invalidArgument('`path` param cannot contain ".."');
+      }
+      return getChild(ref, path);
+    } else {
+      return ref;
+    }
+  }
+}
+
+export function ref(storage: StorageService, url?: string): Reference;
+export function ref(
+  storageOrRef: StorageService | Reference,
+  path?: string
+): Reference;
+/**
+ * Returns a storage Reference for the given url, or given path in the
+ * default bucket.
+ * @param serviceOrRef - `Storage` instance or storage `Reference`.
+ * @param pathOrUrlStorage - path, or URL. If empty, returns root reference (if Storage
+ * instance provided) or returns same reference (if Reference provided).
+ */
+export function ref(
+  serviceOrRef: StorageService | Reference,
+  pathOrUrl?: string
+): Reference | null {
+  if (pathOrUrl && isUrl(pathOrUrl)) {
+    if (serviceOrRef instanceof StorageService) {
+      return refFromURL(serviceOrRef, pathOrUrl);
+    } else {
+      throw invalidArgument(
+        'To use ref(service, url), the first argument must be a Storage instance.'
+      );
+    }
+  } else if (pathOrUrl !== null) {
```

Comment:
```suggestion
  } else if (pathOrUrl) {
```
to be consistent with the if block before.

Reply:
Fixed.